other/upgrade.php

```diff
@@ -1562,7 +1573,52 @@ function DatabaseChanges()
 			// Do we actually need to do this still?
 			if (!isset($modSettings['smfVersion']) || $modSettings['smfVersion'] < $file[1])
 			{
+				// Use STRICT mode on more recent steps
 				setSqlMode($file[3]);
+
+				// Reload modSettings to capture any adds/updates made along the way
+				$request = $smcFunc['db_query']('', '
+					SELECT variable, value
+					FROM {db_prefix}settings',
+					array(
+						'db_error_skip' => true,
+					)
+				);
+				$modSettings = array();
+				while ($row = $smcFunc['db_fetch_assoc']($request))
+					$modSettings[$row['variable']] = $row['value'];
+				$smcFunc['db_free_result']($request);
```

Comment:
To keep things more readable please use a space after a condition/loop is applied:

```php
while ($row = $smcFunc['db_fetch_assoc']($request))
	$modSettings[$row['variable']] = $row['value'];

	$smcFunc['db_free_result']($request);
```

Theres a few whiles and if/elses that can be separated.

Reply:
Think I got 'em.  Any more, let me know.